### PR TITLE
Task 2 cargo getters and c-tor complete

### PR DESCRIPTION
**Pull request by lukasz-kukulka**

### CODE REVIEW

File: shm/inc/Cargo.hpp [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+#include <string>
+class Cargo {
+public:
+    Cargo(std::string, size_t, size_t);
+    std::string getName() const { return name_; }
+    size_t getAmount() const { return amount_; }
+    size_t getBestPrice() const { return basePrice_; }
+
+private:
+    const std::string name_ { };
```
Comment (mariuszlisowski) [inline shm/inc/Cargo.hpp]: No need for curly braces. Default std::string initialization is "" and size of zero.

File: shm/inc/Cargo.hpp [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+class Cargo {
+public:
+    Cargo(std::string, size_t, size_t);
+    std::string getName() const { return name_; }
+    size_t getAmount() const { return amount_; }
+    size_t getBestPrice() const { return basePrice_; }
+
+private:
+    const std::string name_ { }; 
+    size_t amount_ { 0 };
```
Comment (mariuszlisowski) [inline shm/inc/Cargo.hpp]: No need for zero inside curly braces:

{0} == {}

File: shm/src/Cargo.cpp [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#include "Cargo.hpp"
+
+Cargo::Cargo(std::string name, size_t amount, size_t basePrice)
```
Comment (mariuszlisowski) [inline shm/src/Cargo.cpp]: const std::string&

File: shm/inc/Cargo.hpp [hunk 4]
```diff
@@ -0,0 +1,15 @@
+#pragma once
+
+#include <string>
```
Comment (mariuszlisowski) [inline shm/inc/Cargo.hpp]: Line

File: shm/inc/Cargo.hpp [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+class Cargo {
+public:
+    Cargo(std::string, size_t, size_t);
```
Comment (mariuszlisowski) [inline shm/inc/Cargo.hpp]: const std::string&

File: shm/src/Cargo.cpp [hunk 6]
```diff
@@ -0,0 +1,7 @@
+#include "Cargo.hpp"
+
+Cargo::Cargo(std::string name, size_t amount, size_t basePrice) 
+    : name_(name)
+    , amount_(amount)
+    , basePrice_(basePrice)
+{}
```
Comment (mariuszlisowski) [inline shm/src/Cargo.cpp]: Newline?

**Review by mariuszlisowski — CHANGES_REQUESTED** [review]

**Comment by BarTes8** [comment]
Looks good, we will find out about lines and formating.

**Review by mariuszlisowski — APPROVED** [review]

**Pull request merged**